crates/noirc_frontend/src/hir/resolution/resolver.rs

```diff
@@ -884,6 +895,41 @@ impl<'a> Resolver<'a> {
                 }
                 Literal::Integer(integer) => HirLiteral::Integer(integer),
                 Literal::Str(str) => HirLiteral::Str(str),
+                Literal::FmtStr(str) => {
+                    let re = Regex::new(r"\{([\S]+)\}").expect(
+                        "ICE: an invalid regex pattern was used for checking format strings",
+                    );
+                    if re.is_match(&str) {
+                        let mut fmt_str_idents = Vec::new();
+                        for field in re.find_iter(&str) {
+                            let matched_str = field.as_str();
+                            let ident_name = matched_str[1..(matched_str.len() - 1)].to_owned();
```

Comment:
Is it possible to avoid `to_owned` here until it is needed in the last two errors below? That way we don't need it for the normal case.

Reply:
yup it is possible. I have switched it